modules/tinymce/src/core/main/ts/fmt/FormatChanged.ts

```diff
@@ -5,101 +5,167 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
-import { Arr, Cell, Obj } from '@ephox/katamari';
+import { Arr, Cell, Obj, Optional, Type } from '@ephox/katamari';
 
 import Editor from '../api/Editor';
-import Tools from '../api/util/Tools';
+import { FormatEvent } from '../api/EventTypes';
+import { EditorEvent } from '../api/util/EventDispatcher';
+import * as NodeType from '../dom/NodeType';
+import { FormatVars } from './FormatTypes';
 import * as FormatUtils from './FormatUtils';
 import * as MatchFormat from './MatchFormat';
 
 export type FormatChangeCallback = (state: boolean, data: { node: Node; format: string; parents: any }) => void;
 
-export type FormatCallbacks = Record<string, FormatChangeCallback[]>;
+export type RegisteredFormats = Record<string, CallbackGroup>;
 
-export interface FormatData {
-  similar?: boolean;
-  callbacks: FormatChangeCallback[];
+export interface UnbindFormatChanged {
+  unbind: () => void;
 }
 
-export type RegisteredFormats = Record<string, FormatData>;
+interface CustomCallback {
+  readonly state: Cell<boolean>;
+  readonly similar: boolean;
+  readonly vars: FormatVars;
+  readonly callback: FormatChangeCallback;
+}
 
-export interface UnbindFormatChanged {
-  unbind: () => void;
+interface CallbackGroup {
+  // Bundle all callbacks with similar=true and unspecified vars
+  readonly similarState: Cell<boolean>;
+  readonly similarCallbacks: FormatChangeCallback[];
+  // Also bundle all callbacks with similar=false and unspecified vars
+  readonly similarFalseState: Cell<boolean>;
+  readonly similarFalseCallbacks: FormatChangeCallback[];
+  // Then for callbacks that have variables, handle everything separately
+  readonly others: CustomCallback[];
```

Comment:
This is called 2 different things on this line alone: `others` and `custom`. Can we please use the same name throughout? Also IIUC, this is for variable based matches, so maybe we should call it something related to that instead?